iotdb/src/main/java/org/apache/iotdb/db/qp/executor/QueryProcessExecutor.java

```diff
@@ -18,116 +18,604 @@
+      PathErrorException, IOException {
+    return queryRouter.groupBy(paths, aggres, expression, unit, origin, intervals, context);
+  }
+
+  @Override
+  public boolean update(Path path, long startTime, long endTime, String value)
+      throws ProcessorException {
+    String deviceId = path.getDevice();
+    String measurementId = path.getMeasurement();
     try {
-      boolean result = true;
-      MManager mManager = MManager.getInstance();
-      Set<String> pathSet = new HashSet<>();
-      for (Path p : paths) {
-        pathSet.addAll(mManager.getPaths(p.getFullPath()));
+      String fullPath = deviceId + "." + measurementId;
+      if (!mManager.pathExist(fullPath)) {
+        throw new ProcessorException(String.format("Time series %s does not exist.", fullPath));
       }
-      if (pathSet.isEmpty()) {
-        throw new ProcessorException("TimeSeries does not exist and cannot be delete data");
+      mManager.getStorageGroupNameByPath(fullPath);
+      TSDataType dataType = mManager.getSeriesType(fullPath);
+      value = checkValue(dataType, value);
+      storageEngine.update(deviceId, measurementId, startTime, endTime, dataType, value);
+      return true;
+    } catch (PathErrorException e) {
+      throw new ProcessorException(e);
+    }
+  }
+
+  @Override
+  public boolean delete(Path path, long timestamp) throws ProcessorException {
+    String deviceId = path.getDevice();
+    String measurementId = path.getMeasurement();
+    try {
+      if (!mManager.pathExist(path.getFullPath())) {
+        throw new ProcessorException(
+            String.format("Time series %s does not exist.", path.getFullPath()));
       }
-      for (String onePath : pathSet) {
-        if (!mManager.pathExist(onePath)) {
+      mManager.getStorageGroupNameByPath(path.getFullPath());
+      storageEngine.delete(deviceId, measurementId, timestamp);
+      return true;
+    } catch (PathErrorException | StorageEngineException e) {
+      throw new ProcessorException(e);
+    }
+  }
+
+
+  @Override
+  public boolean insert(InsertPlan insertPlan)
+      throws ProcessorException {
+
+    try {
+      String[] measurementList = insertPlan.getMeasurements();
+      String deviceId = insertPlan.getDeviceId();
+      MNode node = mManager.getNodeByDeviceIdFromCache(insertPlan.getDeviceId());
+      String[] values = insertPlan.getValues();
+      TSDataType[] dataTypes = new TSDataType[measurementList.length];
+
+      for (int i = 0; i < measurementList.length; i++) {
+        if (!node.hasChild(measurementList[i])) {
           throw new ProcessorException(
-              String.format("TimeSeries %s does not exist and cannot be delete its data", onePath));
+              String.format("Current deviceId[%s] does not contains measurement:%s",
+                  deviceId, measurementList[i]));
         }
+        MNode measurementNode = node.getChild(measurementList[i]);
+        if (!measurementNode.isLeaf()) {
```

Comment:
a minor improvement: 
you can getChild first, and if getChild() == null, then it means that the timeseries does not exist. (so that you can avoid call hasChild())

Reply:
Calling hasChild() or getChild() does not make much difference in a leaf-node.